R/tests.R

```diff
@@ -124,17 +127,21 @@ renv_tests_diagnostics <- function() {
 
 }
 
-# Cache absolute path to tests/testthat
 renv_tests_root <- function() {
-  global("tests.root", normalizePath(testthat::test_path(".")))
+  `_renv_tests_root` <<- `_renv_tests_root` %||% {
+    normalizePath(testthat::test_path("."), winslash = "/")
+  }
 }
 
 renv_tests_path <- function(path = NULL) {
-  if (is.null(path)) {
-    renv_tests_root()
-  } else {
-    file.path(renv_tests_root(), path)
-  }
+
+  # special case for NULL path
+  if (is.null(path))
```

Comment:
I see this more as working around a buglet in `file.path()`, because IMO `file.path()` should silently drop all `NULL` arguments, but the early return is fine too.

Reply:
I agree; it really feels like there should be a straightforward way to handle this in a single call to `file.path()`. 😞